lerobot/common/datasets/utils.py

```diff
@@ -61,17 +61,35 @@ def unflatten_dict(d, sep="/"):
     return outdict
 
 
-def hf_transform_to_torch(items_dict: dict[torch.Tensor | None]):
+def hf_transform_to_torch(
+    items_dict: dict[torch.Tensor | None],
+    lang_tokenizer_name: str = "t5-small",
+):
     """Get a transform function that convert items from Hugging Face dataset (pyarrow)
     to torch tensors. Importantly, images are converted from PIL, which corresponds to
     a channel last representation (h w c) of uint8 type, to a torch image representation
     with channel first (c h w) of float32 type in range [0,1].
     """
+    # tokenize language instructions if it exists
+    if "language_instruction" in items_dict:
+        from transformers import AutoTokenizer
+
+        tokenizer = AutoTokenizer.from_pretrained(lang_tokenizer_name)
+        tokenizer_kwargs = {
+            "max_length": 64,  # NOTE: adjust this value accordingly
+            "padding": "max_length",
+            "truncation": True,
+            "return_tensors": "pt",
+        }
```

Comment:
Could you add some comments on what this is doing (provides examples?) and explain why we use ` AutoTokenizer.from_pretrained(lang_tokenizer_name)`?

What are the supported `lang_tokenizer_name` ?

Why do we use `t5-small`?

Reply:
Sure I'll add more insight into this. 
This is inspired by how [OpenVLA](https://github.com/openvla/openvla/blob/317dbd9e41edaa83ddf2b33f5412b5e7ddadcd4e/prismatic/models/backbones/llm/base_llm.py#L152) and [OCTO](https://github.com/octo-models/octo/blob/241fb3514b7c40957a86d869fecb7c7fc353f540/octo/data/utils/text_processing.py#L34) process the language data and converts them to language embedding since most OpenX datasets contain a ```language_instruction``` item. 

```t5-small``` is a flexible text-to-text model and the smallest of the T5 models in terms of number of params.

I am not an expert at all so if you have any suggestions regarding this part or if you want to leave it to a future PR, please let me know.

Reply:
This was introduced for language conditioned task. Adding `lang_tokenizer_name` is to support different language encoder.